### DISCUSSION TITLE
how to draw new data from server component in browser?

### DISCUSSION THREAD

**Discussion by dante01yoon** [vercel/next.js#51412] [App Router]
below is how my Post page looks like, I want to re-paint post date time per re fetch data.
I checked my server console output is changed per page refresh, but page doesn't draw fresh data.
<img width="301" alt="스크린샷 2023-06-16 오후 9 35 10" src="https://github.com/vercel/next.js/assets/6510430/d1d7efdf-4ff5-49ed-9d6a-2353dd622e9f">

```tsx

export default async function PostPage() {
  const header = headers()
  console.log({ header })
  console.log("request-time: ", header.get("request-time"))
  const posts = await getPosts()

  return (
    <div className="container mx-auto py-8">
      <h1 className="text-3xl font-bold mb-6">Posts</h1>
      {posts.map((post) => (
        <div key={post.id} className="bg-white p-4 rounded shadow mb-4">
          <h2 className="text-2xl font-bold mb-2">{post.title}</h2>
          <p className="text-gray-600 mb-2">
            By {post.author} | {post.date}
          </p>
          <p className="mb-4">{post.content}</p>
          <div className="bg-gray-100 p-2 rounded">
            {post.comments.map((comment, idx) => (
              <div
                key={comment.id}
                className="mb-2"
                // onClick={handleClick}
              >
                <p className="text-gray-600 mb-1">{comment.author}</p>
                <p>{comment.text}</p>
                <div>{post.time}</div>
              </div>
            ))}
          </div>
        </div>
      ))}
    </div>
  )
}

```

1. How can I force my page draw new data when server fetched data changed ?  ( server re run on page refresh) 
2. How to  force my page draw new data when use soft navigation via next/link 's Link tag? (server doesn't re run)

**Comment by hesalx** [comment]
Check https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config
The caching of `fetch` feels a bit too aggressive in some cases.
